src/__tests__/downshift.get-item-props.js

```diff
@@ -161,32 +168,72 @@ test(`getItemProps doesn't include event handlers when disabled is passed (for I
   }
 })
 
+test(`disabled item can't be selected by pressing enter`, () => {
+  const items = [
+    {item: 'Chess', disabled: true},
+    {item: 'Dominion', disabled: true},
+    {item: 'Checkers', disabled: true},
+  ]
+  const utils = renderDownshift({items})
+  const {input, arrowDownInput, enterOnInput, changeInputValue} = utils
+
+  const firstItem = utils.queryByTestId('item-0')
+  expect(firstItem.hasAttribute('disabled')).toBe(true)
+
+  changeInputValue('c')
+  // ↓
+  arrowDownInput()
+  // ENTER to select the first one
+  enterOnInput()
+  // item was not selected -> input value should still be 'c'
+  expect(input.value).toBe('c')
+})
+
 function renderDownshift({
-  items = ['Chess', 'Dominion', 'Checkers'],
+  items = [{item: 'Chess'}, {item: 'Dominion'}, {item: 'Checkers'}],
   props,
 } = {}) {
-  const renderSpy = jest.fn(({getItemProps}) => (
+  const renderSpy = jest.fn(({getItemProps, getInputProps}) => (
     <div>
+      <input {...getInputProps({'data-testid': 'input'})} />
       {items.map((item, index) => (
         <div
-          {...getItemProps({item, index})}
+          {...getItemProps({
+            ...item,
+            index,
+          })}
           key={index}
           data-testid={`item-${index}`}
         >
-          <button data-testid={`item-${index}-button`}>{item}</button>
+          <button data-testid={`item-${index}-button`}>{item.item}</button>
         </div>
       ))}
     </div>
   ))
-  const utils = render(
+  const utils = renderIntoDocument(
     <Downshift
       isOpen={true}
       onChange={() => {}}
       render={renderSpy}
       {...props}
     />,
   )
-  return {...utils, renderSpy}
+  const input = utils.queryByTestId('input')
+  return {
+    ...utils,
+    renderSpy,
+    input,
+    arrowDownInput: extraEventProps =>
+      fireEvent.keyDown(input, {key: 'ArrowDown', ...extraEventProps}),
+    arrowUpInput: extraEventProps =>
+      fireEvent.keyDown(input, {key: 'ArrowUp', ...extraEventProps}),
+    enterOnInput: extraEventProps =>
+      fireEvent.keyDown(input, {key: 'Enter', ...extraEventProps}),
+    changeInputValue: (value, extraEventProps) => {
+      input.value = value
+      fireEvent.change(input, {...extraEventProps})
```

Comment:
I change the value of the input manually beforehand. Is there a better way to do it like passing it to the event?

Reply:
Maybe, but I prefer this anyway :+1: Thanks!